lib/order.rb

```diff
@@ -0,0 +1,35 @@
+require_relative 'restaurant'
+
+class Order
+
+  attr_reader :total
+
+  def initialize(restaurant_klass: Restaurant)
+    @total = 0
+    @restaurant_klass = restaurant_klass
+    @restaurant = @restaurant_klass.new
+  end
+
+  def see_menu
+    @restaurant.dishes
+  end
+
+  def place_order(item, quantity)
+    item.each do |x|
+      if see_menu.include?(x)
+        q = quantity[item.index(x)]
+        @total += (q*see_menu[x])
+        "you have selected #{x} x#{q}"
+      else
+        raise 'Sorry but your order is currently not on the menu'
```

Comment:
Style/SignalException: Use `fail` instead of `raise` to signal exceptions. (https://github.com/bbatsov/ruby-style-guide#fail-method)